### DISCUSSION TITLE
Can Bevy now be used to create 2D game production environments

### DISCUSSION THREAD

**Discussion by cherish-ltt** [bevyengine/bevy#14722] [Q&A] [answered]
I am currently using godot rust to create 2D games, and I really like the rust language. I would like to try using bevy to complete my work. Can it be used in a production environment now

**Comment by alice-i-cecile** [comment, marked as answer]
Yes, although you're likely to encounter more breaking changes and missing features than you might in Godot :) 2D is relatively mature (even if I grumble about having to use quaternions): both sprite-sheet based and Spine-style animations work well. I would stick to games that either don't need extensive level editing (procedurally generated or puzzles) or are tilemap-based (there's solid editors and integration is easy). The lack of editor tooling can be challenging for content authoring: you'll need to find a workflow that fits your game and 

There's a few areas where you should lean on the ecosystem:

1.  Picking / raycasting: use `bevy_mod_picking`.
2. Audio: use `bevy_kira_audio` or maybe `bevy_fmod`
3. Input binding: use `leafwing-input-manager`
4. Physics: use `avaian` or `bevy_rapier`

The recent [Bevy's 4th Birthday Post](https://bevyengine.org/news/bevys-fourth-birthday/#even-more-real-projectstm-using-bevy) lists a number of production projects, most of which are 2D :) Feedback from those users has largely been positive, although assets, UI and audio are all notably immature.

**Reply by cherish-ltt** [reply]
That's really great, thank you very much. I want to immediately use Bevy to create a 2D game world
